samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,579 @@
+		       ctx->range_start, ctx->range_start + sizeof(ctx->flash_buffer) - 1);
+		return ret;
+	}
+
+	/* Handle HTTP status codes - centralized error management */
+	if (ctx->http_status_code >= 300) {
+		printf("HTTP error %d, requested range=%d-%d\n", ctx->http_status_code,
+		       ctx->range_start, ctx->range_start + sizeof(ctx->flash_buffer) - 1);
+		return -EPROTO;
+	}
+	return ret;
+}
+
+static int ota_update_http_range_header(struct app_ctx *ctx)
+{
+	ctx->range_start += sizeof(ctx->flash_buffer);
+
+	if (ctx->range_start >= (int)ctx->image_size) {
+		return -ENODATA;
+	}
+	return 0;
+}
+
+/*
+ * @brief Process firmware update data packets
+ */
+static int ota_load_firmware(struct app_ctx *ctx)
+{
+	uint32_t ota_calc_image_size = 0;
+	uint32_t status = 0;
+	int buf_offset = 0;
+
+	__ASSERT(ctx->flash_buffer_len > 0, "No firmware data to process");
+
+	if (ctx->range_start == 0) {
+		buf_offset = sizeof(sl_si91x_firmware_header_t);
+		status = sl_wifi_get_firmware_size((void *)ctx->flash_buffer, &ota_calc_image_size);
+		if (status != SL_STATUS_OK) {
+			printf("Unable to fetch firmware size. Status: 0x%x\n", status);
+			return -EIO;
+		}
+
+		if (ctx->image_size != ota_calc_image_size) {
+			printf("Image size mismatch - expected: %u, calculated: %u (file may be "
+			       "corrupted)\n",
+			       ctx->image_size, ota_calc_image_size);
+			return -EINVAL;
+		}
+
+		printf("Firmware size: %u bytes\n", ctx->image_size);
+
+		status = sl_si91x_fwup_start(ctx->flash_buffer);
+		if (status != SL_STATUS_OK) {
+			printf("Failed to load firmware header (0x%x)\n", status);
+			return -EINVAL;
+		}
+	}
+
+	status = sl_si91x_fwup_load(ctx->flash_buffer + buf_offset,
+				    ctx->flash_buffer_len - buf_offset);
+
+	printf("Firmware load success for requested range=%d-%d\n", ctx->range_start,
+	       ctx->range_start + ctx->flash_buffer_len - 1);
+
+	/* Check if firmware update is completed */
+	if (status == SL_STATUS_SI91X_FW_UPDATE_DONE) {
+		printf("Firmware update completed. Rebooting...\n");
+		sys_reboot();
+	}
+	if (status != SL_STATUS_OK) {
+		return -EIO;
+	}
+	return 0;
+}
+
+/*
+ * @brief Main OTA application state machine
+ */
+static int ota_application_start(struct app_ctx *ctx)
+{
+	int ret = 0;
+
+	if (!ctx->connected) {
+		ota_wifi_start_scan(ctx);
+		ret = ota_wifi_connect(ctx);
+		if (ret) {
+			return ret;
+		}
+	}
+
+	if ((IS_ENABLED(CONFIG_NET_IPV4) && !ctx->ipv4_addr_config_done) ||
+	    (IS_ENABLED(CONFIG_NET_IPV6) && !ctx->ipv6_addr_config_done)) {
+		ret = ota_wait_for_ip_config(ctx);
+		if (ret < 0) {
+			return ret;
+		}
+	}
+
+	if (ctx->sock < 0) {
+		ret = ota_connect_to_server(ctx);
+		if (ret < 0) {
+			goto out_close;
+		}
+	}
+
+	ret = ota_download_firmware_chunk(ctx);
+	if (ret < 0) {
+		goto out_close;
+	}
+
+	ret = ota_load_firmware(ctx);
+	if (ret < 0) {
+		goto out_reset;
+	} else {
+		ret = ota_update_http_range_header(ctx);
+		if (ret < 0) {
+			goto out_reset;
+		}
+	}
+	return ret;
```

Comment:
It does not change the behavior, but `return 0;` is easier to understand.

(for the coding style, keep a blank between normal return and error management)